docs/src/model_creation/coupled_crn_equations.md

```diff
@@ -0,0 +1,438 @@
+
+### [Using higher-order differentials](@id coupled_equations_other_higher_order_differentials)
+Until now we have only declared first-order differential equations. However, higher-order differential equations can also be used. These are mostly handled identically to how first-order differential equations are handled, with three exceptions:
+- Variables and differentials are not inferred for these, and must be explicitly declared.
+- Like for [coupled CRN/algebraic equation models](@ref coupled_equations_algebraic_equations), the `structural_simplify = true` option must be provided to `ODEProblem`.
+- Initial conditions must be provided for any lower-order differentials.
+
+The second point will make things a bit more convoluted, as we will need to explicitly define the differential outside of the DSL to set its initial condition.
+
+To demonstrate workflows using higher-order differentials we declare a simple model featuring these:
+```@example coupled_eqs_higher_order_diffs
+using Catalyst # hide
+higher_order_model = @reaction_network begin
+    @variables A(t)
+    @parameters ω
+    @differentials D = Differential(t)
+    @equations begin
+        D(D(A)) + ω*A / (X+1) ~ 0
+    end
+    (p,d), 0 <--> X
+end
+nothing # hide
+```
+Next, we must [@unpack](@ref ref) to retrieve all unknowns (as their initial conditions cannot be provided in `Symbol` form simultaneously as differential initial conditions are provided).
+```@example coupled_eqs_higher_order_diffs
+@unpack A, X = higher_order_model
+nothing # hide
+```
+Next, we define the Differential `D` in our current scope (fetching the same independent variable used with `@reaction_network` from our model using `Catalyst.get_iv`):
+```@example coupled_eqs_higher_order_diffs
+D = Differential(Catalyst.get_iv(higher_order_model))
+nothing # hide
+```
+Finally, we can define our `ODEProblem`, simulate it, and plot the solution:
+```@example coupled_eqs_higher_order_diffs
+u0 = [X => 0.1, A => 1.0, D(A) => 1.0]
+tspan = (0.0, 10.0)
+p = [:p => 1.0, :d => 0.5, :ω => 2.0]
+oprob = ODEProblem(higher_order_model, u0, tspan, p; structural_simplify = true)
+sol = solve(oprob)
+plot(sol)
+```
+We note that while modelling of higher-order coupled CRN/differential equation models using the DSL is a bit convoluted (due to the requirement of re-declaring the differential outside of the DSL), this is not the case for [programmatic modelling](@ref coupled_equations_programmatic). Hence, for these models, this is instead the recommended approach.
+
+## [Coupled CRN/equation models for programmatic modelling](@id coupled_equations_programmatic)
```

Comment:
Right now, I start with going through how coupling work with the DSL. Then, at the end I have a separate section describing the details one needs to change when using a programmatic workflow (here and most other cases it is straightforward, except for the base syntax the two approaches are really similar).

Reply:
How to compose `ODESystem`s with `ReactionSystem`s will be in the compositional modelling part (or at least I think that was our plan when we discussed this a while ago, still happy to change things around)